selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/projectexplorer/UploadIntoProjectTest.java

```diff
@@ -78,37 +79,224 @@ public void selectProjectItem() {
 
   @Test
   public void shouldUploadFileWithDefaultOptions() throws URISyntaxException, IOException {
-    // open upload file window
-    menu.runCommand(
-        TestMenuCommandsConstants.Project.PROJECT, TestMenuCommandsConstants.Project.UPLOAD_FILE);
+    // given
+    final String uploadingFileName = "Aclass.java";
+    final String pathToUploadingFileInsideTheProject =
+        format("%s/%s", PROJECT_NAME, uploadingFileName);
+    final Path localPathToFileToUpload =
+        get(PROJECT_SOURCES.getPath())
+            .resolve("src/main/java/che/eclipse/sample")
+            .resolve(uploadingFileName);
 
-    uploadFileDialogPage.waitOnOpen();
+    // open upload file window
+    openFormAndSelectUploadFile(localPathToFileToUpload);
 
-    // when
-    uploadFileDialogPage.selectResourceToUpload(Paths.get(FILE_TO_UPLOAD_PATH.toURI()));
     uploadFileDialogPage.clickOnUploadButton();
 
     // then
     uploadFileDialogPage.waitOnClose();
     notificationPopup.waitExpectedMessageOnProgressPanelAndClosed(
-        format("File '%s' has uploaded successfully", FILE_TO_UPLOAD_NAME));
-    projectExplorer.waitVisibleItem(format("%s/%s", PROJECT_NAME, FILE_TO_UPLOAD_NAME));
+        format("File '%s' has uploaded successfully", uploadingFileName));
+    projectExplorer.waitVisibleItem(format("%s/%s", PROJECT_NAME, uploadingFileName));
+
+    // Check that uploading file doesn't overwrite existed one
+    // when change the file
+    projectExplorer.openItemByPath(pathToUploadingFileInsideTheProject);
+    editor.typeTextIntoEditor(TEXT_TO_INSERT);
+
+    // when re-upload the file
+    projectExplorer.waitAndSelectItem(PROJECT_NAME);
+
```

Comment:
if steps belong to one logical part "// when re-upload the file" better do not separate they by empty rows

Reply:
You are right, I'll correct